### PR TITLE
Donation link with utm query string

### PR DESCRIPTION
**Pull request by nickhsine**
### Change Reason
https://kanbanflow.com/t/mkh8MzGM

### CODE REVIEW

File: packages/react-components/src/donation-link-with-utm.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const handleClick = e => {
+    e.preventDefault()
+
+    let donationURL = externalLinks.donation
+
+    try {
+      const utmSource = window.location.host
+      const utmCampaign = encodeURIComponent(window.location.pathname)
+      const search = `utm_source=${utmSource}&utm_medium=${utmMedium}&utm_campaign=${utmCampaign}`
+      const url = new URL(donationURL)
```
Comment (YuCJ) [inline packages/react-components/src/donation-link-with-utm.js]: I found that IE and some version of edge do not support the URL object (or not support some properties)
https://developer.mozilla.org/en-US/docs/Web/API/URL

Comment (nickhsine) [reply packages/react-components/src/donation-link-with-utm.js]: I will change to `import URL = require('url')`.

Comment (nickhsine) [reply packages/react-components/src/donation-link-with-utm.js]: Updated. 
Please have a see the [last commit](https://github.com/twreporter/twreporter-npm-packages/pull/11/commits/e37da37ec867de1b23f30060b95539ad363a0426)

**Comment by nickhsine** [comment]
Updated. 
Please have a see.

File: packages/react-components/src/donation-link-with-utm.js (hunk 1 above)
Comment (YuCJ) [reply packages/react-components/src/donation-link-with-utm.js]: But the code is run on browser? Or can we use library like `url-parse` or `qs`?

Comment (YuCJ) [reply packages/react-components/src/donation-link-with-utm.js]: After discussion, the client will take the webpack bundled js file that include the `url` module given by Node. So it's okay to use node modules.

Comment (nickhsine) [reply packages/react-components/src/donation-link-with-utm.js]: `url` would be bundled by `webpack`. 
Hence, the codes could be executed on browser(tested).

**Comment by YuCJ** [comment]
LGTM 👍

**Comment by nickhsine** [comment]
rebase commits.

**Pull request merged**